{"metadata": {"kernelspec": {"display_name": "Bash", "language": "bash", "name": "bash"}, "language_info": {"codemirror_mode": "shell", "file_extension": ".sh", "mimetype": "text/x-sh", "name": "bash"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "markdown", "source": "----\n# A Note on Sample Data\n----", "metadata": {}}, {"cell_type": "markdown", "source": "For the remainder of the notebook, you will be assigned one or more of the following real files from our dataset:\n\n\n|SRA Sample\t|Sample Name|File Name|\n|-----------|-----------|---------|\n|SRS1794108|High-Fat Diet Control 1|SRR5017135.fastq.gz|\n|SRS1794110|High-Fat Diet Control 2|SRR5017137.fastq.gz|\n|SRS1794106|High-Fat Diet Control 3|SRR5017133.fastq.gz|\n|SRS1794105|High-Fat Diet Tumor 1|SRR5017132.fastq.gz|\n|SRS1794101|High-Fat Diet Tumor 2|SRR5017128.fastq.gz|\n|SRS1794111|High-Fat Diet Tumor 3|SRR5017138.fastq.gz|\n\nIn the explanations of the steps we will first use the file `small.fastq.gz`. Then in the exercises portion of the notebook, you will run the analysis on your assigned file(s)\n", "metadata": {}}, {"cell_type": "markdown", "source": "----\n# Running FastQC\n----", "metadata": {}}, {"cell_type": "markdown", "source": "Now, we are ready to analyze our data. Move into the \u2018seqData\u2019 directory in the project folder,\nthis will keep all of our results together, instead of each running it in our own home folder.", "metadata": {}}, {"cell_type": "markdown", "source": "Now let's check the content of our `fastq` folder - these are the pre-imported files we want to do quality checks on.", "metadata": {}}, {"cell_type": "code", "source": "ls /home/gea_user/data/raw_data/fastq", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "As you can see, we have 6 [fastq files](https://en.wikipedia.org/wiki/FASTQ_format) to analyze. To get the data, we will use a program called [fastqc](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/). To run the program we use the command `fastqc` and the name of the file we wish to analyze. We will move into the directory where are data are stored. ", "metadata": {}}, {"cell_type": "code", "source": "cd /home/gea_user/data/raw_data/fastq", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "We will analyze one file to get familiar with the fastqc output (this file is almost 5GB so this may take a few minutes to complete):", "metadata": {}}, {"cell_type": "code", "source": "fastqc small.fastq.gz", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "Our output is returned in two files:\n 1. An .html file (than can be viewed as a webpage)\n 2. A .zip file that contains individual images and reports", "metadata": {}}, {"cell_type": "code", "source": "ls *.html && ls *.zip", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "We can make a new directory to place these results in ", "metadata": {}}, {"cell_type": "code", "source": "mkdir -p /home/gea_user/rna-seq-project/fastqc-untrimmed-results", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "Now let's move these results to the new directory", "metadata": {}}, {"cell_type": "code", "source": "mv *.zip /home/gea_user/rna-seq-project/fastqc-untrimmed-results\nmv *.html /home/gea_user/rna-seq-project/fastqc-untrimmed-results", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "You can browse your HTML (webpage) results in the file browser on the left (rna-seq-project > fastqc-results) - you can click the top-most folder icon to navigate to the home directory for this Jupyter lab session.  \n\n<br><img src=\"./img/folder_nav.png\" alt=\"foldernavigation\" style=\"width:400px;height:275px;\"><br> ", "metadata": {}}, {"cell_type": "markdown", "source": "We will run fastqc on all our files and examine the output in the next notebook:", "metadata": {}}, {"cell_type": "markdown", "source": "----\n# Exercise 2: FASTQ\n----\nAs a reminder - in this laboratory you will be assigned one or more of the 6 FASTQ files to follow through the rest of the analysis. The data files from the leptin experiment are: \n\n\n|SRA Sample\t|Sample Name|File Name|\n|-----------|-----------|---------|\n|SRS1794108|High-Fat Diet Control 1|SRR5017135.fastq.gz|\n|SRS1794110|High-Fat Diet Control 2|SRR5017137.fastq.gz|\n|SRS1794106|High-Fat Diet Control 3|SRR5017133.fastq.gz|\n|SRS1794105|High-Fat Diet Tumor 1|SRR5017132.fastq.gz|\n|SRS1794101|High-Fat Diet Tumor 2|SRR5017128.fastq.gz|\n|SRS1794111|High-Fat Diet Tumor 3|SRR5017138.fastq.gz|", "metadata": {}}, {"cell_type": "markdown", "source": "1. In the blank cell below, run `FastQC` on your assigned sample (e.g. if you are assigned High-Fat Diet Control 1) your filename is `SRR5017135.fastq.gz`. To run the `FastQC` program you type `fastqc`, then a space, then th name of the file:\n\nExampe: \n\n`fastqc fastqfile.fastq.gz`", "metadata": {}}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "2. Use the command below to move the output of the fastqc analysis to the `fastqc-results` folder. We do this using the `mv` (move) command below", "metadata": {}}, {"cell_type": "markdown", "source": "Let's move all results to our previously created folder:", "metadata": {}}, {"cell_type": "code", "source": "mv *.zip /home/gea_user/rna-seq-project/fastqc-untrimmed-results\nmv *.html /home/gea_user/rna-seq-project/fastqc-untrimmed-results", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "markdown", "source": "3. You can view the results in the file browser on the left of the Jupyter-Lab screen; navigate to the `rna-seq-project` folder and then the `fastqc-untrimmed-results` folder. \n\n<br><img src=\"./img/folder_nav.png\" alt=\"foldernavigation\" style=\"width:400px;height:300px;\"><br> ", "metadata": {}}]}